fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/ResourceFactory.java

```diff
@@ -112,4 +112,12 @@ public FedoraResource getResource(final String transactionId, final FedoraId fed
      * @return Stream of child resources
      */
     public Stream<FedoraResource> getChildren(final String transactionId, final FedoraId resourceId);
+
+    /**
+     * Get the interaction model of the resource, assumes you've checked it exists already.
+     * @param transactionId The transaction id
+     * @param fedoraId Identifier of the resource.
+     * @return The interaction model.
+     */
+    public String getInteractionModel(final String transactionId, final FedoraId fedoraId);
```

Comment:
It certainly can work here; however, it seems like "getInteractionModel" would more appropriately be a "FedoraResource" method, rather than a "ResourceFactory" method.

The following is available on "FedoraResourceImpl"
```
psSession.getHeaders(fedoraId, fedoraId.getMementoInstant());
```

Would it make sense for "FedoraLdp" to create the known-to-exist resource, then request its interaction-model?

Reply:
I'm fine with that.